odxtools/compumethods/createanycompumethod.py

```diff
@@ -168,19 +169,27 @@ def create_any_compu_method_from_et(et_element, doc_frags: List[OdxDocFragment],
         return ScaleLinearCompuMethod(linear_methods=linear_methods)
 
     elif compu_category == "TAB-INTP":
-
-        scales = et_element.findall("COMPU-INTERNAL-TO-PHYS/COMPU-SCALES/COMPU-SCALE")
-        internal = [scale.findtext("LOWER-LIMIT") for scale in scales]
-        physical = [scale.findtext("COMPU-CONST/V") for scale in scales]
-
-        internal = [internal_type.from_string(x) for x in internal]
-        physical = [physical_type.from_string(x) for x in physical]
+        internal_points = []
+        physical_points = []
+        for scale_elem in et_element.iterfind("COMPU-INTERNAL-TO-PHYS/COMPU-SCALES/COMPU-SCALE"):
+            internal_point = internal_type.from_string(
+                odxrequire(scale_elem.findtext("LOWER-LIMIT")))
+            physical_point = physical_type.from_string(
+                odxrequire(scale_elem.findtext("COMPU-CONST/V")))
+
+            if not isinstance(internal_point, (float, int)):
+                odxraise()
+            if not isinstance(physical_point, (float, int)):
```

Comment:
What are you basing the assumption that physical and internal points can only be float or int?

Reply:
it is pretty hard to interpolate between anything else? (what is the middle between the strings "foo" and "bar"?)

Reply:
bytes and strings can be compared (for equality and ordering), python natively support that, if they can be compared, then they can be used for table based compu method

Reply:
yes, but the "TAB-INTP" compu methods interpolates between the specified points. e.g. if points (0, 0) and (2, pi) are specified by the PDX file, encountering a 1 while decoding would be mapped to pi/2. (I might be wrong, I haven't actually implemented this code...)